Sources/Task/TaskCollections.swift

```diff
@@ -25,13 +25,16 @@ extension CollectionType where Generator.Element: FutureType, Generator.Element.
         }
 
         let coalescingDeferred = Deferred<Task<Void>.Result>()
-        let outerProgress = NSProgress(parent: nil, userInfo: nil)
-        outerProgress.totalUnitCount = numericCast(count)
+        let progress = NSProgress(parent: nil, userInfo: nil)
+        progress.totalUnitCount = numericCast(count)
         let group = dispatch_group_create()
 
         for task in self {
-            let innerProgress = NSProgress.wrapped(task, cancellation: nil)
-            outerProgress.adoptChild(innerProgress, orphaned: true, pendingUnitCount: 1)
+            if let task = task as? Task<Generator.Element.Value.Value> {
```

Comment:
The method is overall generic over `Collection where Iterator.Element: FutureProtocol, Iterator.Element.Value: Either, Iterator.Element.Value.Left == Error` (Swift 3, sorry), allowing you to intersperse an array of `Task`s and `Future`s. If it is a `Task<T>`, we want to get its progress/cancellation out. If it isn't, then the sub-task can't be cancelled, but you can still attempt to cancel the overall `Task`.